src/Request.js

```diff
@@ -245,17 +246,23 @@ export default class Request {
 
     retry () {
 
-        if (this.retries++ < this._max_retry) {
+        if (this.last_error) {
+            this.errors.push(this.last_error);
+        }
+
+        if (this.retries + 1 < this._max_retry) {
             this.log('warn', 'Retrying request');
+            this.retries++;
             return this.then(this._final_cb);
         }
 
+        const last_error = this.errors.pop();
```

Comment:
Why not do the push after we check if we have reached the max retry or not? Then we don't need to push/pop?